Source/Fields/LMCCylindricalCavity.cc

```diff
@@ -87,33 +87,54 @@ namespace locust
 
     double CylindricalCavity::Z_TE(int l, int m, int n, double fcyc) const
     {
-    	// fix me (placeholder)
     	double Z_TE = 1.0;
     	double x_lm = fInterface->fBesselNKPrimeZeros[l][m];
     	double k1 = x_lm / fInterface->fR;
     	double k3 = n * LMCConst::Pi() / fInterface->fL;
     	double k = pow(k1*k1+k3*k3,0.5);
     	double k0 = fcyc / LMCConst::C();
+    	double A = 1.;
+    	double B = k*k;
+    	double C = k0*k0;
+    	double Q = 1.;  // Q for suppressed modes.
+    	if ((l==0)&&(m==1)&&(n==1)) Q = 1000.; // Q for lmn = 011.
+    	double denom = (Q*B + Q*C + C) * (Q*B+ Q*C + C) + C*C;
+    	double real = (Q*A * (Q*B + Q*C + C)) / denom;
+    	double imag = Q*A*C / denom;
```

Comment:
I get a slightly different set of terms when I rearrange eq. 7.132 to get this. I _think_ that the minus sign on the k_0^2 was dropped in your version. I get:

    	double denom = (Q*B - Q*C - C) * (Q*B - Q*C - C) + C*C;
    	double real = (Q*A * (Q*B - Q*C - C)) / denom;
    	double imag =  -1. * Q*A*C / denom;

with the same solutions in the TM version of this function. Also admittedly the minus on the `imag` term doesn't matter since it's only used when squared. I also may have messed up my own algebra, so it might be worth double checking this.

Reply:
Thank you!  I just recalculated and now I am getting the same answer as you.  I pushed the changes to this branch.